static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -2,179 +2,235 @@ define([
+                        var padding = parseInt($adBannerInner.css('padding-top'))
+                            + parseInt($adBannerInner.css('padding-bottom'));
+                        var clientHeight = parseInt(data.value.height) + padding;
+                        resolve(clientHeight);
                     });
-                };
+                }
+            }
+        });
+    });
+
+    var getLatestAdHeight = function () {
+        var $iframe = getAdIframe();
+        var slotWidth = $iframe.attr('width');
+        var slotHeight = $iframe.attr('height');
+        // iframe may not have been injected at this point
+        var isFluidAd = $iframe.length > 0 && [slotWidth, slotHeight].join(',') === '88,70';
+        // fluid ads are currently always 250px high. We can't just read the client height as fluid ads are
+        // injected asynchronously, so we can't be sure when they will be in the dom
+        var fluidAdInnerHeight = 250;
+        var fluidAdPadding = 18;
+        var fluidAdHeight = fluidAdInnerHeight + fluidAdPadding;
+
+        var adHeightPromise = isFluidAd
+            ? Promise.resolve(fluidAdHeight)
+            : getClientAdHeight();
+        return Promise.race([newRubiconAdHeightPromise, adHeightPromise]);
+    };
 
-                var getAdIframe = function () { return $('iframe', $adBanner); };
-
-                var getLatestAdHeight = function () {
-                    var $iframe = getAdIframe();
-                    var slotWidth = $iframe.attr('width');
-                    var slotHeight = $iframe.attr('height');
-                    // iframe may not have been injected at this point
-                    var isFluidAd = $iframe.length > 0 && [slotWidth, slotHeight].join(',') === '88,70';
-                    // fluid ads are currently always 250px high. We can't just read the client height as fluid ads are
-                    // injected asynchronously, so we can't be sure when they will be in the dom
-                    var fluidAdInnerHeight = 250;
-                    var fluidAdPadding = 18;
-                    var fluidAdHeight = fluidAdInnerHeight + fluidAdPadding;
-
-                    return isFluidAd
-                        ? Promise.resolve(fluidAdHeight)
-                        : getClientAdHeight();
+    var getScrollCoords = function () {
+        return fastdom.read(function () {
+            return [window.pageXOffset, window.pageYOffset];
+        });
+    };
+
+    var getInitialState = function () {
+        var headerHeightPromise = fastdom.read(function () {
+            return $header.height();
+        });
+        return Promise.all([getLatestAdHeight(), headerHeightPromise, getScrollCoords()])
+            .then(function (args) {
+                var adHeight = args[0];
+                var headerHeight = args[1];
+                var scrollCoords = args[2];
+                return {
+                    adIsResizing: false,
```

Comment:
Instead of `adIsResizing`, would `inTransition`, or something like that, be a more accurate name? As the ad itself isn't the thing moving, but the nav itself is moving to compensate for the ad height.